tools/packaging/common/envoy_bootstrap.json

```diff
@@ -494,7 +494,8 @@
           "socket_address": {
             "protocol": "TCP",
             "address": "{{ .wildcard }}",
-            "port_value": 15090
+            "port_value": 15090,
+            "ipv4_compat": true
```

Comment:
So with this set, even when we bind to `::` on a node with no ipv6 support whatsoever, this will work?

Reply:
No, if ipv6 is disabled in the kernel, or not even loaded "::" can't be used (of course). Again, please see;

https://github.com/Nordix/istio/blob/5eaa5d7052462a030bbd6d6885a0d98ed1c0c133/pilot/pkg/networking/core/v1alpha3/listener.go#L2173-L2189

The ipv6 wilcard is only returned when `node.SupportsIPv6()` returns "true".

On the other place I have not done that check, but it is easy to add.

The remaining question is if envoy could choke on  `"ipv4_compat": true` even though the address is IPv4-any. I don't think so but it is worth checking.

Reply:
This line is controlled by https://github.com/istio/istio/pull/29076/files#diff-8f3cc6e4d0a41f3371120cb3c1fcf80af3bc37eb7e0b0f4c696578b75352f0aeR124 even if we don't support ipv6 as far as I can tell?

Reply:
Perhaps an approach where "::" with ipv4_compat is _always_ selected if both ipv4 and ipv6 is supported in the kernel (check the "lo" device for instance) is possible? 0.0.0.0 would _only_ be used if ipv6 is not supported at all.

@howardjohn What do you think?